content/blog/2020/2020-10-01-document-embedders.md

```diff
@@ -0,0 +1,73 @@
+Embedding models mostly bases on neural networks.
+
+Document embedding is computed in two steps. First, each word is embedded with the word embedding then word embeddings are aggregated. The most common type of aggregation is the average over each dimension.
+
+## Why and when should we use embedders?
+
+Comparing to [Bag of Words](/widget-catalog/text-mining/bagofwords-widget), which counts the number of appearances of each token in the document, embeddings have two main advantages:
+
+- **They do not have a dimensionality problem** The result of Bag of Words is a table which has the number of features equal to the number of unique tokens in all documents in a corpus. Large corpora with long texts result in a large number of unique tokens. It results in huge tables which can exceed memory in the computer. Huge tables also increase the learning and evaluation time of machine learning models. Embedders have constant dimensionality of the vector, which is 300 for FastText embeddings that Orange uses. For smaller corpora, we still suggest using Bag of Words since feature space is small.
+- **Most of the preprocessing is not required** In case of Bag of Words approach, we solve the dimensionality problem with the text preprocessing where we remove tokens (e.g. words) that seems to be less important for the analysis. It can also cause the removal of some important tokens. When using embedders, we do not need to remove tokens, so we are not losing the accuracy this way. Also most of the basic preprocessing can be omitted (such as normalization) in case of FastText embeddings.
+
+The shortcoming of the embedders is that they are difficult to understand. For example, when we use a bag-of-words, we can easily observe which tokens are important for classification with [Nomogram widget](https://orange.biolab.si/widget-catalog/visualize/nomogram/) since tokens itself are features. In the case of document embeddings, features are numbers which are not understandable to human by itself.
+
+## Document Embedding widget
+
+Orange now offers document embedders through Document Embedding widget. We decided to use [FastText pretrained embedders](https://fasttext.cc/docs/en/crawl-vectors.html), which support 157 languages. Orange's Document Embedding widget currently supports 31 most common languages.
+
+{{% workflow-screenshot src="/blog_img/2020/2020-10-01-document-embedding-widget.png" %}}
+
+In the widget, the user sets the language of documents and the aggregation method -- it is how embeddings aggregate embeddings for each word in document embedding. 
+
+## The Fake News dataset
+
+For this tutorial, we use the sample of [Fake News datast](). The dataset sample is available at [Orange's file repository](). The dataset is split in two sets training set including 2725 text items and testing set with 275 items. Each item is an article which is labelled as a real or fake.
```

Comment:
Missing data set?

Reply:
The dataset is split in two sets: training set including 2725 text items and testing set with 275 items.
(missing colon)